app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionarySettings.java

```diff
@@ -106,7 +90,27 @@ public void onCreate(Bundle savedInstanceState) {
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.user_dictionary_preference_list_fragment, container, false);
+        LinearLayout view = (LinearLayout) inflater.inflate(R.layout.user_dictionary_preference_list_fragment, container, false);
+
+        final Drawable drawable = ResourcesCompat.getDrawable(getResources(), R.drawable.ic_user_dictionary_add_word, null);
+        // Parameters of the LinearLayout containing the button
+        LinearLayoutCompat.LayoutParams params = new LinearLayoutCompat.LayoutParams(
+                LinearLayoutCompat.LayoutParams.WRAP_CONTENT, LinearLayoutCompat.LayoutParams.WRAP_CONTENT);
+        params.gravity = Gravity.END;
+        params.setMargins(toPixel(0), toPixel(0), toPixel(23), toPixel(10));
+
+        Button addWordButton = new Button(new ContextThemeWrapper(requireContext(), R.style.User_Dictionary_Button), null, 0);
+        addWordButton.setText(R.string.user_dict_add_word_button);
+        addWordButton.setTextColor(getResources().getColor(android.R.color.white));
+        addWordButton.setPadding(toPixel(30), toPixel(10), toPixel(10), toPixel(10));
+        addWordButton.setCompoundDrawablesWithIntrinsicBounds(null, null, drawable, null);
+        addWordButton.setCompoundDrawablePadding(toPixel(20));
+        addWordButton.setLayoutParams(params);
+        addWordButton.setOnClickListener(v -> showAddOrEditDialog(null, null, null));
+
+        view.addView(addWordButton);
```

Comment:
Can you rather add the button in the xml? Having all the UI items in one place is easier to understand, and this way you get a nice preview in Android Studio.

Then you should only need to set the click listener in code.

Same for the list fragment. You could just have an xml with a single button, `include` it in `user_dictionary_preference_list_fragment` and `inflate` it in the list fragment (btw xml name user_dictionary_preference_list_fragment is a bit misleading as it's used for settings instead)

Reply:
Commit 7c1cb10 done

I'm sorry to bother you with this, but I think I did what you asked me to do wrong.
Indeed, there's a white band where the button is, which I can't make transparent.
When you've got some time, could you please tell me what I've done wrong?

<details>
<summary><b>See screenshot</b></summary>

<img width=200 src="https://github.com/Helium314/openboard/assets/139015663/083acf64-c8ba-49d2-bb3c-68804c4559e9">

</details>